Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceAlgorithms/SphericalYeeAlgorithm.H

```diff
@@ -0,0 +1,162 @@
+        amrex::ignore_unused(n_coefs_r);
+
+        amrex::Real const inv_dr = coefs_r[0];
+        amrex::Real const rph = r + 0.5_rt*dr;
+        amrex::Real const rmh = r - 0.5_rt*dr;
+        return 1._rt/(r*r) * inv_dr*( rph*rph*F(i,j,k,comp) - rmh*rmh*F(i-1,j,k,comp) );
+    }
+
+    /** Applies the differential operator `1/r * d(rF)/dr`,
+     * where `F` is on a *cell-centered* grid in `r`
+     * and the differential operator is evaluated on a *nodal* grid.
+     * The input parameter `r` is given at the node-centered position */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real DownwardDrr_over_r (
+        amrex::Array4<amrex::Real const> const& F,
+        amrex::Real const r, amrex::Real const dr,
+        amrex::Real const * const coefs_r, int const n_coefs_r,
+        int const i, int const j, int const k, int const comp ) {
+
+        using namespace amrex::literals;
+        amrex::ignore_unused(n_coefs_r);
+
+        amrex::Real const inv_dr = coefs_r[0];
+        return 1._rt/r * inv_dr*( (r+0.5_rt*dr)*F(i,j,k,comp) - (r-0.5_rt*dr)*F(i-1,j,k,comp) );
+    }
+
+    /**
+     * Perform derivative along r on a cell-centered grid, from a nodal field `F` */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real UpwardDr (
+        amrex::Array4<amrex::Real const> const& F,
+        amrex::Real const * const coefs_r, int const n_coefs_r,
+        int const i, int const j, int const k, int const comp ) {
+
+        amrex::ignore_unused(n_coefs_r);
+
+        amrex::Real const inv_dr = coefs_r[0];
+        return inv_dr*( F(i+1,j,k,comp) - F(i,j,k,comp) );
+    }
+
+    /**
+     * Perform derivative along r on a nodal grid, from a cell-centered field `F` */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real DownwardDr (
+        amrex::Array4<amrex::Real const> const& F,
+        amrex::Real const * const coefs_r, int const n_coefs_r,
+        int const i, int const j, int const k, int const comp ) {
+
+        amrex::ignore_unused(n_coefs_r);
+
+        amrex::Real const inv_dr = coefs_r[0];
+        return inv_dr*( F(i,j,k,comp) - F(i-1,j,k,comp) );
+    }
+
+    /** Applies the differential operator `1/r * d(r * dF/dr)/dr`,
+     * where `F` is on a *cell-centered* or a nodal grid in `r`
+     * The input parameter `r` is given at the cell-centered position */
+    template< typename T_Field>
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real Dr_rDr_over_r (
+        T_Field const& F,
+        amrex::Real const r, amrex::Real const dr,
+        amrex::Real const * const coefs_r, int const /*n_coefs_r*/,
+        int const i, int const j, int const k, int const comp ) {
+
+        using namespace amrex::literals;
+
+        amrex::Real const inv_dr2 = coefs_r[0]*coefs_r[0];
+        return 1._rt/r * inv_dr2*( (r+0.5_rt*dr)*(F(i+1,j,k,comp) - F(i,j,k,comp))
+                                 - (r-0.5_rt*dr)*(F(i,j,k,comp) - F(i-1,j,k,comp)) );
+    }
+
```

Comment:
It looks like this is not currently used in the code ; therefore I would remove it for now.
```suggestion
```